### PR TITLE
Add user customizable prefix for custom targets in cmake

### PR DESCRIPTION
**Pull request by jandrej**
re #1923

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1986](https://github.com/mfem/mfem/pull/1986) | @jandrej | @tzanio | @adrienbernede + @tzanio | 03/24/21 | 03/24/21 | 03/27/21 | |

### CODE REVIEW

**Comment by tzanio** [comment]
@ajkunen, will this work for you instead of #1923?

You can e.g. say `cmake .. -DMFEM_CUSTOM_TARGET_PREFIX="mfem_"`

**Review by ajkunen — APPROVED** [review]
@tzanio yeah that looks great!  Thanks!

**Comment by jandrej** [comment]
bump @tzanio

**Comment by white238** [comment]
We are investigating using MFEM as a submodule then including it in our build system via `add_subdirectory`.  This would help us in that direction.

File: CMakeLists.txt
Comment (adrienbernede) [inline CMakeLists.txt:560]: @jandrej 
When MFEM_CHECK_TARGET_NAME is empty, CMAKE_TEST_COMMAND becomes the target name, and it's the path to ctest binary.
You need a default name for the check target.

Comment (white238) [reply CMakeLists.txt:560]: Line 553 handles this.  It wont be empty. it will either be just `check` or `${MFEM_CUSTOM_TARGET_PREFIX}check`

Comment (adrienbernede) [reply CMakeLists.txt:560]: Yeah, I was right and wrong at the same time :)

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Review by tzanio — APPROVED** [review]

**Comment by tzanio** [comment]
@white238, can you check if that works for you (and approve if so)?

**Comment by white238** [comment]
> @white238, can you check if that works for you (and approve if so)?

@joshessman-llnl is actively working on this so I'll leave that up to him.

**Review by joshessman-llnl — APPROVED** [review]
Confirmed to work on https://github.com/LLNL/serac/compare/tpls/essman/mfem_submodule

**Comment by tzanio** [comment]
@adrienbernede, I assume you approve?

**Review by adrienbernede — APPROVED** [review]

**Comment by adrienbernede** [comment]
LGTM

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**
